.tools/test/stacks/deploy.py

```diff
@@ -1,144 +1,327 @@
-    it outputs the error message and the stack trace.
+    Raises:
+        subprocess.CalledProcessError: If the command execution fails.
     """
-    # Prepare the environment
     env = os.environ.copy()
     if env_vars:
         env.update(env_vars)
 
     command_str = " ".join(command)
-    print("COMMAND: " + command_str)
+    logger.info(f"COMMAND: {command_str}")
+
     try:
         output = subprocess.check_output(command, stderr=subprocess.STDOUT, env=env)
-        print(f"Command output: {output.decode()}")
+        logger.info(f"STDOUT:\n{output.decode()}")
     except subprocess.CalledProcessError as e:
-        print(f"Error executing command: {e.output.decode()}")
+        logger.error(f"Error executing command: {e.output.decode()}")
         raise
     except Exception as e:
-        print(f"Exception executing command: {e!r}")
+        logger.error(f"Exception executing command: {e!r}")
         raise
 
 
-def validate_alphanumeric(value, name):
+def validate_alphanumeric(value: str, name: str) -> None:
     """
     Validate that the given value is alphanumeric.
 
     Args:
-    value (str): The value to validate.
-    name (str): The name of the variable for error messages.
+        value: The value to validate.
+        name: The name of the variable for error messages.
 
     Raises:
-    ValueError: If the value is not alphanumeric.
+        ValueError: If the value is not alphanumeric.
     """
     if not re.match(r"^\w+$", value):
         raise ValueError(f"{name} must be alphanumeric. Received: {value}")
 
 
-def deploy_resources(account_id, account_name, dir, lang="typescript"):
+def get_tokens(account_id: str) -> None:
     """
-    Deploy resources to a specified account using configuration specified by directory and language.
+    Get AWS tokens for the specified account.
 
     Args:
-    account_id (str): The AWS account ID where resources will be deployed.
-    account_name (str): A human-readable name for the account, used for environment variables.
-    dir (str): The base directory containing deployment scripts or configurations.
-    lang (str, optional): The programming language of the deployment scripts. Defaults to 'typescript'.
-
-    Changes to the desired directory, sets up necessary environment variables, and executes
-    deployment commands.
+        account_id: The AWS account ID for which tokens will be obtained.
     """
-    validate_alphanumeric(account_id, "account_id")
-    validate_alphanumeric(account_name, "account_name")
+    get_token_tool = os.getenv("TOKEN_TOOL")
+    get_token_provider = os.getenv("TOKEN_PROVIDER")
 
-    if dir not in os.getcwd():
-        os.chdir(f"{dir}/{lang}")
+    if not all([get_token_tool, get_token_provider]):
+        raise DeploymentError(
+            "TOKEN_TOOL and TOKEN_PROVIDER environment variables must be set"
+        )
 
-    # Securely update tokens
     get_tokens_command = [
-        "ada",
+        get_token_tool,
```

Comment:
No. They were added just to conceal the internal AWS commands that we use, given this is versioned publicly. I believe I explained this in the README, but this may take some tweaking for an external customer to use.